subsys/bluetooth/controller/ll_sw/ull_connections.c

```diff
@@ -223,7 +225,247 @@ void ll_rx_enqueue(struct node_rx_pdu *rx)
 	sys_slist_append(&ut_rx_q, (sys_snode_t *) rx);
 }
 
+/* TODO(tosk): Should be general utility function (used in various places) */
+static uint16_t ull_conn_event_counter(struct ll_conn *conn)
+{
+	struct lll_conn *lll;
+	uint16_t event_counter;
+
+	uint16_t lazy = conn->llcp.prep.lazy;
+
+	lll = &conn->lll;
+
+	/* Calculate current event counter */
+	event_counter = lll->event_counter + lll->latency_prepare + lazy;
+
+	return event_counter;
+}
+
+bool ull_conn_is_params_changed(struct ll_conn *conn, uint16_t interval,
```

Comment:
Perhaps this should be in ull_llcp_conn_upd.c - so as to not 'load' ull_connections.c with cu specific code?
Perhaps we should have a discussion on this on Thursday?

Reply:
Agree! Moved (and renamed)